### PR TITLE
feat: code action for OSS via LS in IntelliJ [IDE-284]

### PR DESCRIPTION
**Pull request by teodora-sandu**
### Description

This PR enhances the existing `codeActions` returned via the LS for OSS with quickfix actions that tell us what version to upgrade a package to. 

The only difference is that the new "OSS via LS" code actions have Snyk Learn actions too and we return quickfix code actions for all possible upgrade paths, rather than just the one.

To review, build https://github.com/snyk/snyk-ls/pull/511 and follow the review instructions there.

### Checklist

- [x] Tests added and all succeed
- [x] Linted
- [ ] README.md updated, if user-facing
- [ ] License file updated, if new 3rd-party dependency is introduced

🚨After having merged, please update the CLI go.mod to pull in latest language server.

### Screenshots / GIFs
This was tested by building `snyk-ls` from https://github.com/snyk/snyk-ls/pull/511 and then the CLI with a `snyk-ls` replace directive, then changing the IntelliJ sandbox generated from this branch to point at the local CLI. Make sure to enable the registry flag.

The tests were run on https://github.com/snyk-labs/snyk-chat-goof.

Before:
Screenshot 2024-05-15 at 10 15 25
Screenshot 2024-05-15 at 10 15 34
Screenshot 2024-05-15 at 10 15 39
Screenshot 2024-05-15 at 10 15 43
Screenshot 2024-05-15 at 10 15 48
Screenshot 2024-05-15 at 10 15 51

After:

Screenshot 2024-05-16 at 16 56 59
Screenshot 2024-05-16 at 16 57 04
Screenshot 2024-05-16 at 16 57 09
Screenshot 2024-05-16 at 16 57 15
Screenshot 2024-05-16 at 16 57 20
Screenshot 2024-05-16 at 16 57 38
Screenshot 2024-05-16 at 16 57 41

### CODE REVIEW

File: src/main/kotlin/snyk/code/annotator/SnykAnnotator.kt [hunk 1]
```diff
@@ -47,25 +51,27 @@ abstract class SnykAnnotator(private val product: ProductType): ExternalAnnotato
                     return@forEach
                 }
                 if (!textRange.isEmpty) {
-                    val annotationMessage = "${issue.annotationMessage()} (Snyk)"
+                    val annotationMessage = issue.annotationMessage()
```
Comment (bastiandoetsch) [inline src/main/kotlin/snyk/code/annotator/SnykAnnotator.kt:54]: What's the reason to remove `(Snyk)`?

**Review by bastiandoetsch — APPROVED** [review]
I suppose there is a corresponding PR that adds it to LS?

**Comment by teodora-sandu** [comment]
Yep @bastiandoetsch it's linked in the PR description too but here it is: https://github.com/snyk/snyk-ls/pull/511. I'll wait to get that merged and released before I merge this PR

File: src/main/kotlin/snyk/code/annotator/SnykAnnotator.kt (hunk 1 above)
Comment (teodora-sandu) [reply src/main/kotlin/snyk/code/annotator/SnykAnnotator.kt:54]: I moved the code that appends it to `src/main/kotlin/snyk/common/lsp/Types.kt`, it makes it a bit more obvious what we render for the different product lines

**Comment by bastiandoetsch** [comment]
The LS change is merged 🚀

**Comment by teodora-sandu** [comment]
Just need to get https://github.com/snyk/cli/pull/5349 merged and then this can be merged!

**Pull request merged**